src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -701,6 +701,17 @@ private void EmitInstrumentedBlock(BoundBlockInstrumentation instrumentation, Bo
 
             if (instrumentation.Epilogue != null)
             {
+                // Check if we're emitting instrumentation try/finally in a catch filter, which produces invalid IL
+                if (_inCatchFilterLevel > 0)
+                {
+                    // Try/finallys are not allowed in catch filters by spec, an error should be reported for this in initial binding.
+                    Debug.Fail("Exception handling constructs should be blocked at the binding layer, not here at emit time");
+
+                    // Report an error as this would produce invalid IL
+                    _diagnostics.Add(ErrorCode.ERR_ModuleEmitFailure, block.Syntax.Location, "Exception handling is not allowed in exception filters");
```

Comment:
>, "Exception handling is not allowed in exception filters"); [](http://example.com/codeflow?start=91&length=60)

It looks like this error requires two arguments. Other places provide module name as the first argument
```
((Cci.INamedEntity)moduleBeingBuilt).Name
```